File: lab10_C#/ReservationGrpc/model/Journey.cs
using System;
using System.Runtime.CompilerServices;

namespace Reservations.model
{
    [Serializable]

    public class Journey : IHasID<string>
    {
        private string id;
        private string touristicObjective;
        private string transportCompany;
        private string departureTime;
        private double price;
        private int noAvailableSeats;

        public Journey(string id, string touristicObjective, string transportCompany, double departureTime, double price, int noAvailableSeats)
        {
            this.touristicObjective = touristicObjective;
            this.transportCompany = transportCompany;

            this.price = price;
            this.noAvailableSeats = noAvailableSeats;
            ID = id;

            string[] splits = departureTime
                .ToString()
                .Split('.');
            if (splits.Length > 1)
            {
                this.departureTime = splits[0] + ":" + splits[1].Substring(0, 2);

            }
            else
            {
                this.departureTime = splits[0] + ":00";
            }
        }

        public string ID
        {
            get { return id; }
            set { id = value; }
        }

        public string TouristicObjective
        {
            get => touristicObjective;
            set => touristicObjective = value;
        }

        public string TransportCompany
        {
            get => transportCompany;
            set => transportCompany = value;
        }

        public String DepartureTime
        {
            get => departureTime;
            set => departureTime = value;
        }

        public double Price
        {
            get => price;
            set => price = value;
        }

        public int NoAvailableSeats
        {
            get => noAvailableSeats;
            set => noAvailableSeats = value;
        }


        public override string ToString()
        {
            return string.Format("[Journey: Id={0}, touristicObjective={1}, transportCompany={2}, departureTime={3}, price={4}, " +
                                 "noAvailableSeats={5}]", id, touristicObjective, transportCompany, departureTime, price, noAvailableSeats);
        }

    }
}

File: lab10/reservationModel/src/main/java/model/Journey.java
package model;

import model.HasID;

import java.io.Serializable;
import java.time.LocalTime;

public class Journey implements HasID<String>, Serializable {
    private String id;
    private String touristiObjective;
    private String transportCompany;
    private LocalTime departureTime;
    private double price;
    private int noAvailableSeats;

    public Journey(String id, String touristiObjective, String transportCompany, LocalTime departureTime, double price, int noAvailableSeats) {
        this.id = id;
        this.touristiObjective = touristiObjective;
        this.transportCompany = transportCompany;
        this.departureTime = departureTime;
        this.price = price;
        this.noAvailableSeats = noAvailableSeats;
    }

    @Override
    public String getID() {
        return id;
    }

    @Override
    public void setID(String newID) {
        id= newID;
    }

    public String getTouristiObjective() {
        return touristiObjective;
    }

    public void setTouristiObjective(String touristiObjective) {
        this.touristiObjective = touristiObjective;
    }

    public String getTransportCompany() {
        return transportCompany;
    }

    public void setTransportCompany(String transportCompany) {
        this.transportCompany = transportCompany;
    }

    public LocalTime getDepartureTime() {
        return departureTime;
    }

    public void setDepartureTime(LocalTime departureTime) {
        this.departureTime = departureTime;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getNoAvailableSeats() {
        return noAvailableSeats;
    }

    public void setNoAvailableSeats(int noAvailableSeats) {
        this.noAvailableSeats = noAvailableSeats;
    }

    @Override
    public String toString() {
        return "model.Journey{" +
                "id='" + id + '\'' +
                ", touristiObjective='" + touristiObjective + '\'' +
                ", transportCompany='" + transportCompany + '\'' +
                ", departureTime=" + departureTime +
                ", price=" + price +
                ", noAvailableSeats=" + noAvailableSeats +
                '}';
    }
}
